### PR TITLE
feat(etls): add interreg 2014TC16RFCB047 - EUBFR-258

### PR DESCRIPTION
**Pull request by kalinchernev**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

When you add a new ETL/producer, please check for the following:

- [ ] Ensure the ETL has been correctly setup following the guide in `services/ingestion/etl/README.md`
- [ ] There is at least 1 sample `record.json` and 1 unit test with a jest snapshot for the result of the transform function of the ETL.
- [ ] Ensure there is (flow/jsdocs) documentation in the transform function
- [ ] Generate the necessary documentation pages for the new ETL by `yarn docs:md`

### CODE REVIEW

File: services/ingestion/etl/2014tc16rfcb047/xls/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+| eMS Ref Num                   | description       |
+| Beneficiary/Lead Partner Name | third_parties     |
+| Operation/Project Name        | title             |
+| Operation Start Date          | timeframe.from    |
+| Operation End Date            | timeframe.to      |
+| Operation/Project Summary     | description       |
+| Committed Outputs             | description       |
+| Total Project Cost (€)        | budget.total_cost |
+| Total ERDF + Match (€)        | description       |
+| Total ERDF Allocated (€)      | description       |
```
Comment (yhuard) [inline services/ingestion/etl/2014tc16rfcb047/xls/README.md]: `Total ERDF Allocated` = `budget.eu_contrib`, you don't need to use `Union Co-Financing Rate %` to recalculate the EU Contribution ;-)

**Review by yhuard — CHANGES_REQUESTED** [review]
Deployed and tested, it works fine. 👍 There's 1 possible improvement, please see my comment below.

File: services/ingestion/etl/2014tc16rfcb047/xls/test/unit/lib/__snapshots__/transform.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  "programme_name": "",
+  "project_id": "045",
+  "project_locations": Array [
+    Object {
+      "address": "Northern Ireland, Ireland & Scotland",
+      "centroid": null,
+      "country_code": "",
+      "location": null,
+      "nuts": Array [],
+      "postal_code": "A91 K584",
```
Comment (yhuard) [inline services/ingestion/etl/2014tc16rfcb047/xls/test/unit/lib/__snapshots__/transform.spec.js.snap]: seems like `A91 K584` is in Dundalk, Ireland, yet the Excel file says `Northern Ireland, Ireland & Scotland`. I don't know what's the best way to extract the `project_locations`. Has Laurent said anything about it?

File: services/ingestion/etl/2014tc16rfcb047/xls/README.md (hunk 1 above)
Comment (kalinchernev) [reply services/ingestion/etl/2014tc16rfcb047/xls/README.md]: Right! Changed!

File: services/ingestion/etl/2014tc16rfcb047/xls/test/unit/lib/__snapshots__/transform.spec.js.snap (hunk 2 above)
Comment (kalinchernev) [reply services/ingestion/etl/2014tc16rfcb047/xls/test/unit/lib/__snapshots__/transform.spec.js.snap]: You're right, I've assumed too much to put countries into address and hope for post code to help.
I changed the logic to use the country names as much as possible to deduct country codes and remove the post code. In this particular example, Scotland and Northern Ireland are not deducted, but Ireland is, so we store its country code which is more precise than the assuming-address with the postcode.

File: services/ingestion/etl/2014tc16rfcb047/xls/src/lib/transform.js [hunk 3]
```diff
@@ -53,9 +49,10 @@ const getBudget = record => {
  * - `Committed Outputs`
  * - `Total ERDF + Match (€)`
  * - `Total ERDF Allocated (€)`
+ * - `Union Co-Financing Rate %`
  * - `Category of Intervention`
  *
- * @memberof 2014tc16rfcb014CsvTransform
+ * @memberof 2014tc16rfcb047XlsTransform
```
Comment (yhuard) [inline services/ingestion/etl/2014tc16rfcb047/xls/src/lib/transform.js]: damn, it's not easy to catch those ones 🤣

File: services/ingestion/etl/2014tc16rfcb047/xls/src/lib/transform.js [hunk 4]
```diff
@@ -103,33 +101,60 @@ const getProjectId = record =>
...
-  locations.push({
-    address: country,
-    centroid: null,
-    country_code: '',
-    location: null,
-    nuts: [],
-    postal_code: postCode,
-    region: '',
-    town: '',
+  const countriesMulti = record.Country.split(',')
```
Comment (yhuard) [inline services/ingestion/etl/2014tc16rfcb047/xls/src/lib/transform.js]: what about just `const countriesMulti = record.Country.split(/\s*(?:,|&)\s*/);`? 😃 

it splits when it finds `,` or `&`, with the surrounding spaces (if any) (equivalent to trim)

the only difference is that it doesn't trim the beginning or the end of the input string

I guess `const countriesMulti = record.Country.trim().split(/\s*(?:,|&)\s*/);` would be a more exact equivalent...

**Review by yhuard — APPROVED** [review]

**Review by yhuard — APPROVED** [review]

**Pull request merged**